src/hazelcore/Action_Pat.re

```diff
@@ -1102,10 +1213,25 @@ and ana_perform_operand =
     Succeeded(
       Statics_Pat.ana_fix_holes_z(ctx, u_gen, body |> place_cursor, ty),
     );
-
+  | (Backspace, CursorP(OnDelim(_ /* 0 */, After), TypeAnn(_, op, _))) =>
+    Succeeded(
+      Statics_Pat.ana_fix_holes_z(
+        ctx,
+        u_gen,
+        op |> ZPat.place_after_operand |> ZOpSeq.wrap,
+        ty,
+      ),
+    )
   /* Construct */
   | (Construct(SOp(SSpace)), CursorP(OnDelim(_, After), _)) =>
     ana_perform_operand(ctx, u_gen, MoveRight, zoperand, ty)
+  | (Construct(SAsc), CursorP(_)) =>
```

Comment:
on dev, this causes the expected type to be inserted immediately. I think that's a useful default, can we retain that behavior?